tools/build-toolchain.sh

```diff
@@ -1,34 +1,74 @@
 #! /bin/bash
-# N64 MIPS GCC toolchain build/install script for Unix distributions
+# N64 MIPS GCC toolchain build/install script (runs under UNIX systems).
 # (c) 2012-2021 Shaun Taylor and libDragon Contributors.
 # See the root folder for license information.
 
 
-
-# Before calling this script, make sure you have GMP, MPFR and TexInfo
-# packages installed in your system.  On a Debian-based system this is
-# achieved by typing the following commands:
+# Before calling this script, make sure you have all required
+# dependency packages installed in your system.  On a Debian-based systems
+# this is achieved by typing the following commands:
 #
-# sudo apt-get install libmpfr-dev
-# sudo apt-get install texinfo
-# sudo apt-get install libmpc-dev
+# sudo apt-get update && sudo apt-get upgrade
+# sudo apt-get install -yq wget bzip2 gcc g++ make file libmpfr-dev libmpc-dev zlib1g-dev texinfo git gcc-multilib
+
+# # Bash strict mode http://redsymbol.net/articles/unofficial-bash-strict-mode/
+# set -euo pipefail
+# IFS=$'\n\t'
 
-# Exit on error
+# Exit script on error
 set -e
 
-# Set N64_INST before calling the script to change the default installation directory path
-INSTALL_PATH="${N64_INST:-/usr/local}"
-# Set PATH for newlib to compile using GCC for MIPS N64 (pass 1)
-export PATH="$PATH:$INSTALL_PATH/bin"
+# Ensure you set 'N64_INST' before calling the script to change the default installation directory path
+  # by default it will presume 'usr/local/n64_toolchain'
+  INSTALL_PATH="${N64_INST:-/usr/local/n64_toolchain}"
+
+# Check for cross compile script flag
+# TODO: detect the flag `--host`
+# TODO: does not reach bash strict as $1 could be null!
+if [ "$1" == "x86_64-w64-mingw32" ]; then # Windows cross compile (host) flag is specified as a parameter.
```

Comment:
I think I prefer that this is not a command line option. If you define HOST/BUILD/TARGET at the global script level like this:

```bash
BUILD=$(BUILD:-x86_64-linux-gnu}
HOST=${HOST:-x86_64-linux-gnu}
TARGET=${TARGET:-mips64-elf}
```

then you can simply override `HOST` externally, such as `HOST=x86_64-w64-mingw32 build-toolchain.sh`

Reply:
Done.